src/main/kotlin/io/emeraldpay/dshackle/upstream/Multistream.kt

```diff
@@ -316,6 +320,23 @@ abstract class Multistream(
         log.info("State of ${chain.chainCode}: height=${height ?: '?'}, status=[$statuses], lag=[$lag], weak=[$weak]")
     }
 
+    @EventListener
+    fun onUpstreamChange(event: UpstreamChangeEvent) {
+        val chain = event.chain
+        if (this.chain == chain) {
```

Comment:
we can use @EventListener(condition = "....")

Reply:
I haven't found a nice looking way to replace it with SpEl :(